src/main/java/org/verapdf/wcag/algorithms/semanticalgorithms/ContrastRatioChecker.java

```diff
@@ -0,0 +1,177 @@
+package org.verapdf.wcag.algorithms.semanticalgorithms;
+
+import org.verapdf.wcag.algorithms.entities.enums.TextType;
+
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.util.*;
+import java.util.List;
+
+public class ContrastRatioChecker {
+
+    public boolean isTextContrastRatioCompliant(BufferedImage sourceTextImage) {
+        return isTextContrastRatioCompliant(sourceTextImage, TextType.REGULAR);
+    }
+    public boolean isTextContrastRatioCompliant(BufferedImage sourceTextImage, TextType type) {
+        return isTextContrastRatioCompliant(sourceTextImage, type, false);
+    }
+
+    public boolean isTextContrastRatioCompliant(BufferedImage sourceTextImage, TextType type, boolean isHighVisibility) {
+
+        List<DataPoint> localMaximums = findLocalMaximums(getLuminosityPresenceList(sourceTextImage));
+
+        System.out.println("Luminance local maximums:");
+        for (int i = 0; i < localMaximums.size(); i++) {
+            System.out.println(i + ": " + localMaximums.get(i).value + " --- " + localMaximums.get(i).totalOccurrence);
+        }
+
+        double [] contrastColors = get2MostPresentElements(localMaximums);
```

Comment:
How do we make sure that 2 most present elements are text and background colors? It is possible that 2 most present elements are both background or both text colors. Also if we render an image with the text we may asume that we know text color from the font metadata, so the remaining problem is to extract background color or background mean color.

Reply:
As discussed during the meeting, support for text input param will be added during actual algorithm implementation